### DISCUSSION TITLE
Error message with nlmixr and RxODE re. CMT

### DISCUSSION THREAD

**Discussion by krinaj** [nlmixrdevelopment/nlmixr#542] [Q&A]
Hello,

What does this error message mean and how can I fix it? I have neither of these mentioned items in my dataset columns. I tried with both with and without cmt columns, but issue persist. 

Error in .fitFun(.ret) :  can only specify either 'cmt', 'ytype', 'state' or 'var'

I am trying to pass time-varying covariates into dataset in nlmixr. The same issue was coming into RxODE when used same dataset to simulate.

Thanks,
Krina

**Comment by mattfidler** [comment]
Hi,  it means that you are trying to use different columns to specify the same thing and RxODE does not know how to translate your dataset.  

If you have `YTYPE` or `DVID` that doesn't sync with the corresponding `CMT` definition this can com e up as well.

**Comment by krinaj** [comment]
Hi Matt,

Thanks so much for your quick reply.

I do not have cmt, dvid, or ytype in my dataset that I am trying to use. I
am just using one type of observations/endpoint. The issue seems to be
coming from covariates in the dataset. If I set a covariate value within
the model code (not variable by time and id in the dataset), the error
message does not come. Is there an example that you can point me to for how
to input covariate info in dataset for nlmixr? I looked at this one, but it
does not fully apply to my example.
https://nlmixrdevelopment.github.io/RxODE/articles/RxODE-covariates.html

Thanks,
Krina

**Reply by mattfidler** [reply]
Perhaps you could list the column names that you have in your dataset.

**Comment by krinaj** [comment]
Hi Matt,

Yeah, sure. Here is what I have in the dataset. I am using the following
code in nlmixr. The killx and statx parameters come from a separate PK
model run previously and I am using them as time varying covariates in the
model I am trying to estimate a few parameters from.

> colnames(dat1)
[1] "id"     "time"   "dv"     "GROUP"  "killE"  "killI"  "killiE" "killiI"
"killiP" "killiR" "killP"  "killR"  "MET"    "statE"  "statI"  "statiE"
"statiI" "statiP" "statiR" "statP"
[21] "statR"  "evid"   "mdv"    "amt"

fit <- nlmixr(f, data=dat1, est="focei", foceiControl(sigdig=3, print=10))

thanks,
Krina

**Comment by krinaj** [comment]
Hi,

I reduced the model as well as the number of covariates columns, and the
error message is gone. It seems like I may have had some errors in the
code.

Thank you and kind regards,
Krina

**Reply by mattfidler** [reply]
Great I'm glad to hear it;  From the column names you posted, I couldn't see why there should be an error
